tests/test_polaraveraging.py

```diff
@@ -0,0 +1,225 @@
+"""
+Tests for polaraveraging.py
+"""
+import clmm
+import clmm.polaraveraging as pa
+from numpy import testing
+import numpy as np
+from astropy.table import Table
+import os
+
+def test_make_bins():
+    testing.assert_equal(len( pa.make_bins(1,10,9,False)),10 )
+    testing.assert_allclose( pa.make_bins(1,10,9,False) , np.arange(1.,11.) )
+    testing.assert_allclose( pa.make_bins(1,10000,4,True) ,10.**(np.arange(5)) )
+    
+    testing.assert_raises(TypeError, pa.make_bins, rmin='glue', rmax=10, n_bins=9, log_bins=False)
+    testing.assert_raises(TypeError, pa.make_bins, rmin=1, rmax='glue', n_bins=9, log_bins=False)
+    testing.assert_raises(TypeError, pa.make_bins, rmin=1, rmax=10, n_bins='glue', log_bins=False)
+    testing.assert_raises(TypeError, pa.make_bins, rmin=1, rmax=10, n_bins=9, log_bins='glue')
+
+    testing.assert_raises(ValueError, pa.make_bins, rmin=1, rmax=10, n_bins=-4, log_bins=False)
+    testing.assert_raises(ValueError, pa.make_bins, rmin=1, rmax=-10, n_bins=9, log_bins=False)
+    testing.assert_raises(ValueError, pa.make_bins, rmin=1, rmax=10, n_bins=0, log_bins=False)
+    testing.assert_raises(TypeError, pa.make_bins, rmin=1, rmax=10, n_bins=9.9, log_bins=False)
+
+def test_compute_g_x():
+    data = np.array([[0.01, 0.02, 0.01], # g1
+                     [0.01, 0.02, 0.03], # g2
+                     [0.1, 1.2, 3.]]) # phi [rad]
+
+    # test that function works for scalar and vector input
+    assert(isinstance(pa._compute_g_x(*data[:,0]), float))
+    assert(isinstance(pa._compute_g_x(*data), np.ndarray))
+    testing.assert_equal(3, len(pa._compute_g_x(*data)))
+    testing.assert_equal(pa._compute_g_x(*(data[:,0])), pa._compute_g_x(*data)[0])
+    
+    # test same length array input
+    testing.assert_raises(ValueError, pa._compute_g_x, data[0,0], data[1], data[2])
+    testing.assert_raises(ValueError, pa._compute_g_x, data[0], data[1,0], data[2])
+    testing.assert_raises(ValueError, pa._compute_g_x, data[0], data[1], data[2,0])
+    testing.assert_raises(ValueError, pa._compute_g_x, data[0,0], data[1,0], data[2])
+    testing.assert_raises(ValueError, pa._compute_g_x, data[0], data[1,0], data[2,0])
+    testing.assert_raises(ValueError, pa._compute_g_x, data[0,0], data[1], data[2,0])
+    
+    # test for input range
+    testing.assert_raises(ValueError, pa._compute_g_x, 0.1, 0.1, -3.15)
+    testing.assert_raises(ValueError, pa._compute_g_x, 0.1, 0.1, 2.*np.pi+0.1)
+    testing.assert_almost_equal(pa._compute_g_x(0.1, 0.1, 0.), pa._compute_g_x(0.1, 0.1, np.pi))
+
+    
+    # test for reasonable values
+    testing.assert_almost_equal(pa._compute_g_x(100., 0., 0.), 0.0)
+    testing.assert_almost_equal(pa._compute_g_x(0., 100., np.pi/4.), 0.0)
+    testing.assert_almost_equal(pa._compute_g_x(0., 0., 0.3), 0.)
```

Comment:
Lets add some reasonable value tests. The math to compute the cross shear is super simple. We can choose same reasonable values, break out a calculator and compute the expectation. Lets do this for two tuples of phi, g1, g2